catalog/ftp/src/main/java/ddf/catalog/ftp/ftplets/FtpRequestHandler.java

```diff
@@ -151,65 +232,118 @@ public FtpletResult onUploadStart(FtpSession session, FtpRequest request)
         }
 
         session.write(new DefaultFtpReply(FtpReply.REPLY_150_FILE_STATUS_OKAY,
-                STOR_REQUEST + " " + fileName));
+                requestTypeString + " " + fileName));
         LOGGER.debug("Replying to client with code 150 - file status okay");
 
-        try (FileBackedOutputStream outputStream = new FileBackedOutputStream(1000000);
-                final AutoCloseable ac = outputStream::reset) {
-            DataConnection dataConnection = connFactory.openConnection();
-            dataConnection.transferFromClient(session, outputStream);
+        if (isDotFile(request.getArgument())) {
+            DataConnection dataConnection;
+            try {
+                dataConnection = connFactory.openConnection();
+            } catch (Exception e) {
+                LOGGER.debug("Error getting the output data stream from the FTP session", e);
+                throw new IOException("Error getting the output stream from FTP session");
```

Comment:
Ultimately, the Exception types needs to be wrapped in an FtpException or IOException because of how the methods are declared by Ftplet. So, it's either wrap it here, or wrap it one stack frame higher. I will change it so the IOException contains the Exception so it doesn't get lost.